TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -120,22 +120,22 @@ class MetAPI{
             }
     }
 
-    func departments(completion: @escaping (DepartmentsResponse?) -> Void) {
+    func departments(completion: @escaping (Result<DepartmentsResponse, MetAPIError>) -> Void) {
         self.metAPICache.departments { [weak self] departmentsResponce in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIdoesNotAnswer))
                 return
             }
             if let departmentsResponce = departmentsResponce {
-                completion(departmentsResponce)
+                completion(.success(departmentsResponce))
             } else {
                 self.executeDepartments {[weak self] realResponse in
                     guard let realResponse = realResponse else {
-                        completion(nil)
+                        completion(.failure(.noDataInResponce))
```

Comment:
You still rely on `nil`/`non-nil` state of the `realResponse`. Please introduce `Result` to `func executeDepartments` as well